File: Jiho/week8/day37/k진수에서 소수 개수 구하기.js
function isPrime(num) {
  if (num <= 1) return false;
  for (let i = 2; i <= Math.floor(Math.sqrt(num)); i++) {
    if (num % i == 0) return false;
  }
  return true;
}

function solution(n, k) {
  let answer = 0;

  // 1. 정수 n을 k진수로 변환한다.
  let NToK = n.toString(k);
  // 2. 변환된 수를 0을 기준으로 나눈다. 그러면 조건에 맞는 수만 반환된다.
  let numList = NToK.split("0").map((v) => +v);
  // 3. 소수인 것들 카운트
  return numList.filter((v) => isPrime(v)).length;
}

File: Jin/week08/day37/k진수에서소수개수구하기.java
package Jin.week08.day37;

public class k진수에서소수개수구하기 {
    static int answer = 0;

    public int solution(int n, int k) {
        // 진수 변환
        String convertNum = Integer.toString(n, k);
        String[] convertArr = convertNum.split("0");

        // 소수가 몇 개인지 확인하기
        check(convertArr);

        return answer;
    }

    private void check(String[] arr) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i].equals("")) continue;

            long number = Long.parseLong(arr[i]);
            if (isPrime(number)) answer++;
        }
    }

    private boolean isPrime(long number) {
        // 1은 소수가 아님
        if (number == 1) return false;

        for (int i = 2; i <= Math.sqrt(number); i++) {
            if (number % i == 0) return false;
        }

        return true;
    }
}
